app/bundles/PluginBundle/EventListener/LeadSubscriber.php

```diff
@@ -29,7 +35,7 @@ public static function getSubscribedEvents(): array
      */
     public function onLeadDelete(LeadEvent $event): bool
     {
-        /** @var \Mautic\LeadBundle\Entity\Lead $lead */
+        /** @var Lead $lead */
         $lead                  = $event->getLead();
         $integrationEntityRepo = $this->pluginModel->getIntegrationEntityRepository();
         $integrationEntityRepo->findLeadsToDelete('lead%', $lead->getId());
```

Comment:
Shouldn't we do same here? Same  in onCompanyDelete

Reply:
@dadarya0 did you have time to check it?

Reply:
@kuzmany we should not restrict record deletion as I can explain in below example.

- Configure any relevant plugin, enable sync
- After sync some record will be added in `integration_entity` table
- Now disable/unpublish plugin
- No if we are restricting deletion of record in `integration_entity` table then those record will be available forever and those are useless and unmapped as lead/company is already deleted.

I believe that now it is clear the reason of not implementing same logic in `onCompanyDelete` and `onLeadDelete`

Reply:
Thanks. Sounds clear 🥇